crates/matrix-sdk-ui/tests/integration/timeline/edit.rs

```diff
@@ -122,3 +129,75 @@ async fn edit() {
     assert_matches!(edited.in_reply_to(), None);
     assert!(edited.is_edited());
 }
+
+#[async_test]
+async fn send_edit() {
+    let room_id = room_id!("!a98sd12bjh:example.org");
+    let (client, server) = logged_in_client().await;
+    let event_builder = EventBuilder::new();
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let mut sync_builder = SyncResponseBuilder::new();
+    sync_builder.add_joined_room(JoinedRoomBuilder::new(room_id));
+
+    mock_sync(&server, sync_builder.build_json_sync_response(), None).await;
+    let _response = client.sync_once(sync_settings.clone()).await.unwrap();
+    server.reset().await;
+
+    let room = client.get_room(room_id).unwrap();
+    let timeline = room.timeline().await;
+    let (_, mut timeline_stream) =
+        timeline.subscribe_filter_map(|item| item.as_event().cloned()).await;
+
+    let event_id = event_id!("$original_event");
+    sync_builder.add_joined_room(JoinedRoomBuilder::new(room_id).add_timeline_event(
+        event_builder.make_sync_message_event_with_id(
+            // Same user as the logged_in_client
+            user_id!("@example:localhost"),
+            event_id,
+            RoomMessageEventContent::text_plain("Hello, World!"),
+        ),
+    ));
+
+    mock_sync(&server, sync_builder.build_json_sync_response(), None).await;
+    let _response = client.sync_once(sync_settings.clone()).await.unwrap();
+    server.reset().await;
+
+    let hello_world_item =
+        assert_next_matches!(timeline_stream, VectorDiff::PushBack { value } => value);
+    let hello_world_message = hello_world_item.content().as_message().unwrap();
+    assert!(!hello_world_message.is_edited());
+    assert!(hello_world_item.can_be_edited());
+
+    mock_encryption_state(&server, false).await;
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/send/.*"))
+        .respond_with(
+            ResponseTemplate::new(200).set_body_json(json!({ "event_id": "$edit_event" })),
+        )
+        .expect(1)
+        .mount(&server)
+        .await;
+
+    timeline
+        .edit(RoomMessageEventContent::text_plain("Hello, Room!"), &hello_world_item)
+        .await
+        .unwrap();
+
+    let edit_item =
+        assert_next_matches!(timeline_stream, VectorDiff::Set { index: 0, value } => value);
+
+    // The event itself is already known to the server. We don't currently have
+    // a separate edit send state.
+    assert_matches!(edit_item.send_state(), None);
+    let edit_message = edit_item.content().as_message().unwrap();
+    assert_eq!(edit_message.body(), "Hello, Room!");
+    assert!(edit_message.is_edited());
+
+    // The response to the mocked endpoint does not generate further timeline
+    // updates, so just wait for a bit before verifying that the endpoint was
+    // called.
+    _ = timeout(std::future::pending::<()>(), Duration::from_millis(200)).await;
```

Comment:
...that's a contrived way to `tokio::time::sleep(Duration::from_millis(200).await`?